### DISCUSSION TITLE
Fluent Logging addKeyValue() vs Parameterized logging

### DISCUSSION THREAD

**Discussion by SidB3** [qos-ch/slf4j#442] [Q&A]
Hello,

I was wondering if I do `LOGGER.atDebug().addKeyValue("key", computeValue()).log("message");`

Will the `computeValue()` method be evaluated if LOGGER is at `error` level or will it work like Parameterized logging and check the log level first?

**Comment by ppkarwasz** [comment]
Sure, `computeValue()` will be eagerly evaluated. To evaluate it lazily you need to pass `Supplier` to `addKeyValue`:

```java
LOGGER.atDebug().addKeyValue("key", this::computeValue).log("message");
```
